editor.planx.uk/src/@planx/components/MapAndLabel/Public/index.test.tsx

```diff
@@ -84,44 +96,194 @@ describe("Basic UI", () => {
+    expect(secondTabPanel.childElementCount).toBeGreaterThan(0);
+    expect(firstTabPanel.childElementCount).toBe(0);
+
+    const continueButton = getByRole("button", { name: /Continue/ });
+    await user.click(continueButton);
+
+    // error messages appear
+    const errorMessagesTabTwo =
+      within(secondTabPanel).getAllByTestId(/error-message-input/);
+    expect(errorMessagesTabTwo).toHaveLength(4);
+
+    // error messages are empty but visible before error state induced
+    // this ensures they contain the error message text
+    errorMessagesTabTwo.forEach((input) => {
+      expect(input).not.toBeEmptyDOMElement();
+    });
+
+    // click to go to the second tab
+    await user.click(firstTab);
+
+    expect(firstTabPanel).toBeVisible();
+
+    // error messages persist
+    const errorMessagesTabOne =
+      within(firstTabPanel).getAllByTestId(/error-message-input/);
+    expect(errorMessagesTabOne).toHaveLength(4);
+  });
+
+  // it shows all fields are required across different tabs
+  it("should show an error if the minimum number of items is not met", async () => {
+    const { getByTestId, getByRole, user } = setup(<MapAndLabel {...props} />);
+    const map = getByTestId("map-and-label-map");
+    expect(map).toBeInTheDocument();
+
+    const continueButton = getByRole("button", { name: /Continue/ });
+
+    await user.click(continueButton);
+
+    const errorWrapper = getByTestId(/error-wrapper/);
+
+    const errorMessage = within(errorWrapper).getByText(/You must plot /);
+
+    expect(errorMessage).not.toBeEmptyDOMElement();
+  });
+  // ??
+  it("an error state is applied to a tabpanel button, when it's associated feature is invalid", async () => {
+    const { getByTestId, getByRole, user, getAllByTestId } = setup(
+      <MapAndLabel {...props} />,
+    );
+    const map = getByTestId("map-and-label-map");
+    expect(map).toBeInTheDocument();
+
+    addFeaturesToMap(map, [point1]);
+
+    const tabOne = getByRole("tab", { name: /Tree 1/ });
+
+    expect(tabOne).toBeInTheDocument();
+
+    const continueButton = getByRole("button", { name: /Continue/ });
+    expect(continueButton).toBeInTheDocument();
+    await user.click(continueButton);
+
+    const errorMessages = getAllByTestId(/error-message-input/);
+
+    // check error messages are correct amount and contain info
+    expect(errorMessages).toHaveLength(4);
+
+    errorMessages.forEach((message) => {
+      expect(message).not.toBeEmptyDOMElement();
+    });
+
+    expect(tabOne).toHaveStyle("border-left: 5px solid #D4351C");
+
+    addFeaturesToMap(map, [point1, point2]);
+
+    const tabTwo = getByRole("tab", { name: /Tree 2/ });
+
+    expect(tabTwo).toBeInTheDocument();
+
+    expect(tabOne).not.toHaveStyle("border-left: 5px solid #D4351C");
+
+    await user.click(continueButton);
+
+    expect(tabOne).toHaveStyle("border-left: 5px solid #D4351C");
+    expect(tabTwo).toHaveStyle("border-left: 5px solid #D4351C");
```

Comment:
Then from here, I am retesting the error state being applied again.

@jessicamcinchak just wanted to check this is the intended functionality for now or is it likely to change soon?

Reply:
See above comment - I think we just need to test error style of a single tab to satisfy this test.